app/controllers/application_controller.rb

```diff
@@ -0,0 +1,14 @@
+class ApplicationController < ActionController::Base
+  # Prevent CSRF attacks by raising an exception.
+  # For APIs, you may want to use :null_session instead.
+  def index
+    @markets = Market.all
+  end
+
+  def show
+    @market = Market.find(params[:id])
+    @vendors = Vendor.where(market_id: @market.id)
+  end
+
+  protect_from_forgery with: :exception
```

Comment:
Consider leaving configuration/validation code at the top of the file just as general style etiquette.